### PR TITLE
Changed consoles to use .vv files + refactoring

### PR DESCRIPTION
**Pull request by suomiy**
This change is [Reviewable](https://reviewable.io/reviews/matobet/movirt/203)

### CODE REVIEW

File: moVirt/src/main/java/org/ovirt/mobile/movirt/rest/Console.java [hunk 1]
```diff
@@ -1,26 +1,19 @@
...
 
 @JsonIgnoreProperties(ignoreUnknown = true)
 public class Console implements RestEntityWrapper<org.ovirt.mobile.movirt.model.Console> {
     public String id;
-    public String address;
     public String protocol;
-    public String port;
-    public String tls_port;
-    public Vm vm;
+    public org.ovirt.mobile.movirt.rest.v4.Vm vm;
```
Comment (suomiy) [inline moVirt/src/main/java/org/ovirt/mobile/movirt/rest/Console.java]: Make rest.vm not abstract and use it here

File: moVirt/src/main/java/org/ovirt/mobile/movirt/rest/VvFileHttpMessageConverter.java [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+    @Override
+    protected Object readInternal(Class<? extends Object> clazz,
+                                  HttpInputMessage inputMessage) throws IOException, HttpMessageNotReadableException {
+        Object result = null;
+        InputStream inputStream = inputMessage.getBody();
+        try {
+            result = convertStreamToConsoleConnectionDetails(inputStream);
+        } finally {
+            if (inputStream != null) {
```
Comment (jelkosz) [inline moVirt/src/main/java/org/ovirt/mobile/movirt/rest/VvFileHttpMessageConverter.java]: can the inputStream be null? If yes, than what will happen to convertStreamToConsoleConnectionDetails if you pass it null?

File: moVirt/src/main/java/org/ovirt/mobile/movirt/rest/VvFileHttpMessageConverter.java [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+            Matcher matcher = p.matcher(sc.nextLine());
+            if (matcher.matches()) {
+                MatchResult result = matcher.toMatchResult();
+                vvFileMap.put(result.group(1), result.group(2));
+            }
+        }
+        ConsoleProtocol protocol = ConsoleProtocol.mapProtocol(vvFileMap.get(VvFileParam.TYPE));
+        String address = vvFileMap.get(VvFileParam.HOST);
+        String password = vvFileMap.get(VvFileParam.PASSWORD);
+        String subject = vvFileMap.get(VvFileParam.HOST_SUBJECT);
```
Comment (jelkosz) [inline moVirt/src/main/java/org/ovirt/mobile/movirt/rest/VvFileHttpMessageConverter.java]: The address, passsword and subject will be null according to this code. Is it ok? Can the rest of the code survive it?

File: moVirt/src/main/java/org/ovirt/mobile/movirt/rest/VvFileHttpMessageConverter.java (hunk 2 above)
Comment (suomiy) [inline moVirt/src/main/java/org/ovirt/mobile/movirt/rest/VvFileHttpMessageConverter.java]: NPE will occur, that means whole request will fail and an error window will be shown with this NPE

Comment (suomiy) [inline moVirt/src/main/java/org/ovirt/mobile/movirt/rest/VvFileHttpMessageConverter.java]: should we throw more specific exception?

File: moVirt/src/main/java/org/ovirt/mobile/movirt/rest/VvFileHttpMessageConverter.java (hunk 3 above)
Comment (suomiy) [inline moVirt/src/main/java/org/ovirt/mobile/movirt/rest/VvFileHttpMessageConverter.java]: if password is set to null, following toast will be shown: Password cannot be null
I guess we should make the same kind of errors also for address and subject, right? 
Now it just sends invalid params to aSpice

File: moVirt/src/main/java/org/ovirt/mobile/movirt/rest/VvFileHttpMessageConverter.java [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+
+    @Override
+    protected Object readInternal(Class<? extends Object> clazz,
+                                  HttpInputMessage inputMessage) throws IOException, HttpMessageNotReadableException {
+        Object result = null;
+        InputStream inputStream = inputMessage.getBody();
+        try {
+            result = convertStreamToConsoleConnectionDetails(inputStream);
+        } catch (Exception x) {
+            throw new IllegalStateException("Couldn't parse .vv file response", x);
```
Comment (suomiy) [inline moVirt/src/main/java/org/ovirt/mobile/movirt/rest/VvFileHttpMessageConverter.java]: fixed

File: moVirt/src/main/java/org/ovirt/mobile/movirt/ui/vms/VmDetailActivity.java [hunk 5]
```diff
@@ -454,6 +462,10 @@ private String makeConsoleUrl(ConsoleConnectionDetails details)
             throw new IllegalArgumentException("Password cannot be null");
         }
 
+        if (StringUtils.isEmpty(details.getAddress())) {
+            throw new IllegalArgumentException("Address cannot be null");
```
Comment (suomiy) [inline moVirt/src/main/java/org/ovirt/mobile/movirt/ui/vms/VmDetailActivity.java]: fixed here

File: moVirt/src/main/java/org/ovirt/mobile/movirt/ui/vms/VmDetailActivity.java [hunk 6]
```diff
@@ -462,6 +474,10 @@ private String makeConsoleUrl(ConsoleConnectionDetails details)
             case SPICE:
                 parameters = Constants.PARAM_SPICE_PWD + "=" + details.getPassword(); // spice password
                 if (details.getTlsPort() > 0) {
+                    if (StringUtils.isEmpty(details.getCertificateSubject())) {
+                        throw new IllegalArgumentException("Certificate subject cannot be null");
```
Comment (suomiy) [inline moVirt/src/main/java/org/ovirt/mobile/movirt/ui/vms/VmDetailActivity.java]: and here

**Comment by suomiy** [comment]
added better error messages

**Pull request merged**